### PR TITLE
Handle unexpected responses gracefully

### PR DESCRIPTION
**Pull request by AndyWendt**
Today, I noticed an interesting issue where I received a `404` response with no `ErrorCode` or `Message` fields in the body.  This broke the exception handling in the code.  The cause I believe was that I tried to send a batch that was too large in size (not in number).  

Anyway, for the fun stuff.

1. Extracted a function, using PHPStorm built-in refactorings, to handle responses
2. Promoted, using PHPStorm built-in refactorings, said function to a class
3. Backfilled tests and, along the way, renamed and restructured the code
4. Extracted factory methods to internalize the knowledge of the construction of `PostmarkException`
5. Handled potentially missing `ErrorCode` and or `Message` keys in the response body by using the PHP 7.0 compatible null coalescing operator. 

There are a couple of defaults that need maintainer input.

### CODE REVIEW

File: src/Postmark/Models/PostmarkException.php [hunk 1]
```diff
@@ -9,6 +9,42 @@ class PostmarkException extends \Exception {
...
-}
 
-?>
\ No newline at end of file
+    public static function from($response)
+    {
+        $ex = new self();
+        $body = json_decode($response->getBody(), true);
+        $ex->httpStatusCode = $response->getStatusCode();
+        $ex->postmarkApiErrorCode = $body['ErrorCode'] ?? 'invalid-response';
```
Comment (AndyWendt) [inline src/Postmark/Models/PostmarkException.php:18]: What should be the default here?

File: src/Postmark/Models/PostmarkException.php [hunk 2]
```diff
@@ -9,6 +9,42 @@ class PostmarkException extends \Exception {
...
+        $ex->message = $body['Message'] ?? 'This appears to be an invalid response. Please contact support.';
```
Comment (AndyWendt) [inline src/Postmark/Models/PostmarkException.php:19]: What should be the default here?

File: src/Postmark/PostmarkClientBase.php [hunk 3]
```diff
@@ -52,13 +52,13 @@ abstract class PostmarkClientBase {
...
 
-	protected function __construct($token, $header, $timeout = 30) {
+    protected function __construct($token, $header, $timeout = 30) {
 		$this->authorization_header = $header;
 		$this->authorization_token = $token;
 		$this->version = phpversion();
 		$this->os = PHP_OS;
 		$this->timeout = $timeout;
-	}
+    }
```
Comment (AndyWendt) [inline src/Postmark/PostmarkClientBase.php:61]: Looks like my IDE did an autoformat

**Comment by manavo** [comment]
@ewood-ac any chance this could be reviewed? I occasionally get errors like this:

```
Trying to access array offset on value of type null at /.../vendor/wildbit/postmark-php/src/Postmark/PostmarkClientBase.php:181)
```

Which I think happens when the response body is not valid JSON, but isn't handled very well. And if I'm not mistaken, that's what this PR tries to resolve?

**Comment by ewood-ac** [comment]
I tried to make this work. But there were lots of failing tests and incompatibilities. So, I opted instead to try and solve this by the simplest means - just handle the null. This fix should go out v6.0.5 in the next couple of days.

**Pull request closed without merging**